assets/src/components/AssignmentTable.js

```diff
@@ -92,16 +104,27 @@ function AssignmentTable (props) {
     handleInputBlur
   } = props
 
-  const [popoverEl, setPopoverEl] = useState({ popoverId: null, anchorEl: null })
+  const assignmentStatus = {
+    GRADED: 'Graded',
+    SUBMITTED: 'Not Yet Graded',
+    UNSUBMITTED: 'Unsubmitted'
+  }
+
+  const assignmentStatusNames = [assignmentStatus.GRADED, assignmentStatus.SUBMITTED, assignmentStatus.UNSUBMITTED]
+  const [filteredAssignments, setFilteredAssignments] = useState(assignments)
```

Comment:
I see that all the some of functions/const is using `filteredAssignments` instead of `assignments`. Can you explain what exactly the idea behind it. Since I can see `assignment` and `filteredAssignments` use approperiately)

Reply:
#1139  was because the keys being used by the functions passed along to AssignmentTable were using indexes into the array of assignments as a 'key' for performing updates.  When the list was filtered inplace the keys started pointing to the wrong array index.  To fix this I switched to using the assignment id as the key.  

However there are several helper functions in AssignmentTable.js that explicitly reference consecutive array indices and I wanted to keep them functionality unchanged so I had used the filtered copy of the array.

Not sure if that really explains it well or not.

Reply:
Can you please add this in the comment? It make other people who step into the code easier if they have little bit of context